src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedEntryPointSymbol.cs

```diff
@@ -451,7 +451,7 @@ internal override BoundBlock CreateBody(BindingDiagnosticBag diagnostics)
                 // Creates a new top-level binder that just contains the global imports for the compilation.
                 // The imports are required if a consumer of the scripting API is using a Task implementation 
                 // that uses extension methods.
-                Binder binder = WithUsingNamespacesAndTypesBinder.Create(compilation.GlobalImports, next: new BuckStopsHereBinder(compilation), withImportChainEntry: true);
+                Binder binder = WithUsingNamespacesAndTypesBinder.Create(compilation.GlobalImports, next: new BuckStopsHereBinder(compilation, syntax.SyntaxTree), withImportChainEntry: true);
```

Comment:
>syntax.SyntaxTree [](http://example.com/codeflow?start=143&length=17)

Do we have a test that shows the impact of having a non-null associated syntax tree in this binder?

It seems a bit irregular that we use a non-null tree here, in contrast with the other two locations I commented on. What is the intent?

Reply:
I misunderstood what this binder does. It doesn't bind user code, it only binds call and await to the entry point of the user's script. Therefore I think it's better to just pass a null syntaxTree here.